src/NuGet.Clients/NuGet.CommandLine/Commands/ProjectFactory.cs

```diff
@@ -735,6 +710,31 @@ private Packaging.Core.PackageDependency CreateDependencyFromProject(dynamic pro
             }
         }
 
+        private static void ExtractMetadata(ProjectFactory projectFactory, Packaging.PackageBuilder builder)
```

Comment:
@alpaix  It is also used to invoke the ExtractMetadataFromProject function . In the second occurence at line 698, the projectFactory is created from the dependency and is different from "this".